### ISSUE TITLE
Temporary raster layer not storable by a click on the temporary layer symbol

### ISSUE THREAD

**Issue by tschmetzer** [qgis/QGIS#43700]
Other than common with vector layers temporary raster layers are not storable by a click on  the temporary layer symbol:
![grafik](https://user-images.githubusercontent.com/67278094/121693002-fb1a3200-cac8-11eb-94e5-66e540f2caf3.png)

Steps for reproduction:
1. Extract raster example file and shape mask file from [raster_example.zip](https://github.com/qgis/QGIS/files/6638709/raster_example.zip) and drag layers into a new project

2. Find and processing algorithm "clip raster by mask layer" and select the respective layers
![grafik](https://user-images.githubusercontent.com/67278094/121690631-5f87c200-cac6-11eb-85be-fca267ae74ce.png)

3. Start processing
![grafik](https://user-images.githubusercontent.com/67278094/121691029-d3c26580-cac6-11eb-81be-0781c994dfac.png)

4. Try saving the newly created temporary layer by clicking on the symbol as shown in the screenshot (doesn't work)
![grafik](https://user-images.githubusercontent.com/67278094/121692553-7af3cc80-cac8-11eb-8260-17300f46b2f8.png)

Only a right click on layer -> "Export/Save as..." works where a separate layer is inserted into the Layers list.

System under test:

QGIS version | 3.18.3-Zürich | QGIS code revision | 735cc85be9
-- | -- | -- | --
Compiled against Qt | 5.11.2 | Running against Qt | 5.11.2
Compiled against GDAL/OGR | 3.1.4 | Running against GDAL/OGR | 3.1.4
Compiled against GEOS | 3.8.1-CAPI-1.13.3 | Running against GEOS | 3.8.1-CAPI-1.13.3
Compiled against SQLite | 3.29.0 | Running against SQLite | 3.29.0
PostgreSQL Client Version | 11.5 | SpatiaLite Version | 4.3.0
QWT Version | 6.1.3 | QScintilla2 Version | 2.10.8
Compiled against PROJ | 6.3.2 | Running against PROJ | Rel. 6.3.2, May 1st, 2020
OS Version | Windows 10 (10.0)
Active python plugins | GroupStats; nominatim; QuickOSM; db_manager; processing

**Comment by gioman** [comment]
@tschmetzer I don't think that is has ever been implemented for raster temporary layers, if so this would be a feature request, not a bug.

**Comment by alexbruy** [comment]
Duplicates #32929.
